profiler/profiler.go

```diff
@@ -492,8 +428,8 @@ func (p *profiler) interruptibleSleep(d time.Duration) {
 func (p *profiler) stop() {
 	p.stopOnce.Do(func() {
 		close(p.exit)
-		p.telemetry.Stop()
 	})
+	telemetry.GlobalClient.Stop()
```

Comment:
Hmm, if someone stopped the profiler but not the tracer, would this stop the tracer's telemetry as well? Perhaps we should send a "product change" message to say that the profiler is disabled instead?

Thinking ahead to having either of the profiler or tracer start/stop telemetry, perhaps this shared client also needs to keep track of which products are using it, and only fully shut down once all of them are stopped. I'm open to deferring that to later, and just assume that only the tracer is going to start & stop telemetry for now.

Reply:
Thanks for catching this! For now I'll remove this line and have the tracer be responsible for starting/stopping telemetry. And yep, I was planning to implement what you described in a future PR (having something in the telemetry client keep track of where it's being used, and only shutdown when it's not being used anywhere).

Reply:
Would sending a "product disabled" message still makes sense? Like `telemetry.GlobalClient.ProductEnabled(telemetry.NamespaceProfilers, false, nil)`

Reply:
Sorry forgot to respond to that portion of the message - I think that makes sense! Just going to double check that product enablement status is correlated with both time and runtime_id, and not just runtime_id (since we don't want the product disabled message to overwrite the enablement status of the product for that runtime).